read.go

```diff
@@ -298,6 +286,32 @@ func (r *reader) readIndexData(toc *indexTOC) (*indexData, error) {
 	return &d, nil
 }
 
+func (r *reader) readMetadata(toc *indexTOC) (*Repository, *IndexMetadata, error) {
```

Comment:
Shall we update ReadMetadata to call this readMetadata?

Reply:
I think you might have missed this, we do :)